File: web-hrd/lib/services/ruleService.js
const API_URL = process.env.NEXT_PUBLIC_API_URL;

const getHeaders = () => {
  const token = localStorage.getItem("token");
  return {
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  };
};

export const ruleService = {
  getRules: async () => {
    const response = await fetch(`${API_URL}/rules`, {
      headers: getHeaders(),
    });
    return response.json();
  },
  updateRules: async (data) => {
    const response = await fetch(`${API_URL}/rules`, {
      method: "PUT",
      headers: getHeaders(),
      body: JSON.stringify(data),
    });
    return response.json();
  },
};

File: mobile_karyawan/lib/data/services/rule_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../../core/constants/app_constants.dart';
import 'auth_service.dart';

class RuleService {
  final String _baseUrl = AppConstants.baseUrl;
  final AuthService _authService = AuthService();

  Future<Map<String, String>> _getHeaders() async {
    final token = await _authService.getToken();
    return {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer $token',
    };
  }

  /// Fetch attendance rules dari backend
  /// Returns: { tolerance_minutes: int, absent_after_minutes: int }
  Future<Map<String, dynamic>?> getRules() async {
    try {
      final response = await http.get(
        Uri.parse('$_baseUrl/rules'),
        headers: await _getHeaders(),
      );
      if (response.statusCode == 200) {
        return jsonDecode(response.body) as Map<String, dynamic>;
      }
    } catch (e) {
      print('Error fetching rules: $e');
    }
    return null;
  }
}
